src/auto/cell_layout.rs

```diff
@@ -94,9 +97,31 @@ impl<O: IsA<CellLayout>> CellLayoutExt for O {
     //    unsafe { TODO: call ffi::gtk_cell_layout_set_attributes() }
     //}
 
-    //fn set_cell_data_func<'a, P: IsA<CellRenderer>, Q: Into<Option<&'a /*Unimplemented*/CellLayoutDataFunc>>, R: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(&self, cell: &P, func: Q, func_data: R, destroy: /*Unknown conversion*//*Unimplemented*/DestroyNotify) {
-    //    unsafe { TODO: call ffi::gtk_cell_layout_set_cell_data_func() }
-    //}
+    fn set_cell_data_func<P: IsA<CellRenderer>, Q: Fn(&CellLayout, &CellRenderer, &TreeModel, &TreeIter) + 'static, R: Into<Option<Q>>>(&self, cell: &P, func: R) {
+        let func = func.into();
+        let func_data: Box_<Option<Q>> = Box::new(func.into());
+        unsafe extern "C" fn func_func<P: IsA<CellRenderer>, Q: Fn(&CellLayout, &CellRenderer, &TreeModel, &TreeIter) + 'static>(cell_layout: *mut ffi::GtkCellLayout, cell: *mut ffi::GtkCellRenderer, tree_model: *mut ffi::GtkTreeModel, iter: *mut ffi::GtkTreeIter, data: glib_ffi::gpointer) {
+            let cell_layout = from_glib_borrow(cell_layout);
+            let cell = from_glib_borrow(cell);
+            let tree_model = from_glib_borrow(tree_model);
+            let iter = from_glib_borrow(iter);
+            let callback: &Option<Q> = &*(data as *mut _);
+            if let Some(ref callback) = *callback {
+                callback(&cell_layout, &cell, &tree_model, &iter)
+            } else {
+                panic!("cannot get closure...")
+            };
+        }
+        let func = if func_data.is_some() { Some(func_func::<P, Q> as _) } else { None };
+        unsafe extern "C" fn destroy_func<P: IsA<CellRenderer>, Q: Fn(&CellLayout, &CellRenderer, &TreeModel, &TreeIter) + 'static>(data: glib_ffi::gpointer) {
+            let _callback: Box_<Q> = Box_::from_raw(data as *mut _);
+        }
+        let destroy_call4 = Some(destroy_func::<P, Q> as _);
+        let super_callback0: Box_<Option<Q>> = func_data;
```

Comment:
Conflict with `let _callback: Box_<Q> = Box_::from_raw(data as *mut _);` in destroy_func

Reply:
Well, it does the same if the value is not `None` in this case. If it's `None` you might free a `NULL` pointer instead :)

Reply:
I still don't get it...

Reply:
It's `Box<Q>` in the destroy notify but you create a `Box<Option<Q>>` and pass that to it. The types don't match. In the closure trampoline the types match

Reply:
Ooooooh!

Reply:
Fixed.

Reply:
Big thanks, we near final 🚀